cmd/private-org-sync/main.go

```diff
@@ -460,35 +460,73 @@ func main() {
 		git:                  gitExec,
 		failOnNonexistentDst: o.failOnNonexistentDst,
 	}
+	var errs []error
+	var locations []location
+
+	whitelist := o.WhitelistOptions.WhitelistConfig.Whitelist
+	if len(whitelist) > 0 {
+		whitelistedLocations, err := getWhitelistedLocations(whitelist, syncer.git)
+		if err != nil {
+			errs = append(errs, err)
+		}
+		locations = append(locations, whitelistedLocations...)
+	}
 
-	var syncErrors []error
 	callback := func(_ *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
-		syncer.logger = config.LoggerForInfo(*repoInfo)
-		source := location{
+		locations = append(locations, location{
 			org:    repoInfo.Org,
 			repo:   repoInfo.Repo,
 			branch: repoInfo.Branch,
-		}
+		})
+		return nil
+	}
+
+	if err := config.OperateOnCIOperatorConfigDir(o.configDir, o.makeFilter(callback)); err != nil {
+		errs = append(errs, err)
+	}
+
+	for _, source := range locations {
+		syncer.logger = config.LoggerForInfo(config.Info{
+			Org:    source.org,
+			Repo:   source.repo,
+			Branch: source.branch,
+		})
+
 		destination := source
 		destination.org = o.targetOrg
 		gitDir, err := syncer.makeGitDir(source.org, source.repo)
 		if err != nil {
-			syncErrors = append(syncErrors, fmt.Errorf("%s->%s: %v", source.String(), destination.String(), err))
-			return nil
+			errs = append(errs, fmt.Errorf("%s->%s: %v", source.String(), destination.String(), err))
+			continue
 		}
 
 		if err := syncer.mirror(gitDir, source, destination); err != nil {
-			syncErrors = append(syncErrors, fmt.Errorf("%s->%s: %v", source.String(), destination.String(), err))
+			errs = append(errs, fmt.Errorf("%s->%s: %v", source.String(), destination.String(), err))
 		}
-		return nil
 	}
 
-	callback = o.makeFilter(callback)
-	if err := config.OperateOnCIOperatorConfigDir(o.configDir, callback); err != nil || len(syncErrors) > 0 {
-		if err != nil {
-			syncErrors = append(syncErrors, err)
+	if len(errs) > 0 {
+		logrus.WithError(utilerrors.NewAggregate(errs)).Fatal("There were failures")
+	}
+}
+
+func getWhitelistedLocations(whitelist map[string][]string, git gitFunc) ([]location, error) {
```

Comment:
I would like to see this tested. There is a git mock already in `main_test.go` which should make testing this not that hard.

Reply:
Well, I don't think that there is a good reason to test that func since it is only iterating in a map and calls `getRemoteBranchHeads`. But instead, I believe that adding unittests for `getRemoteBranchHeads` should cover the required testing.

Reply:
Ah this is already being tested in `TestMirror`.